llvm/lib/Transforms/Scalar/Scalarizer.cpp

```diff
@@ -1348,5 +1350,6 @@ PreservedAnalyses ScalarizerPass::run(Function &F, FunctionAnalysisManager &AM)
   bool Changed = Impl.visit(F);
   PreservedAnalyses PA;
   PA.preserve<DominatorTreeAnalysis>();
+  PA.preserve<DXILResourceAnalysis>();
```

Comment:
I'm not sure if this line and line 355 are ok. If there was never a `DXILResourceWrapperPass` initialized before this pass then there should be nothing to preserve right?  If not Is there any other way to do this? We've done a bunch of abstraction to keep the DirectX specific out of this pass.

Reply:
If there was never a DXILResourceWrapperPass initialized then preserving it is a no-op. This is needed so that *if* the DXILResourceWrapperPass ran before this pass we don't need to recalculate it again *after* this pass. If we don't do this then we need to run resource analysis twice in our pipeline.